pyiceberg/transforms.py

```diff
@@ -323,6 +327,16 @@ def hash_func(v: Any) -> int:
 
                 return mmh3.hash(struct.pack("<q", v))
 
+        elif isinstance(source, (TimestampNanoType, TimestamptzNanoType)):
+
+            def hash_func(v: Any) -> int:
+                if isinstance(v, py_datetime.datetime):
+                    v = datetime.datetime_to_micros(v)
```

Comment:
I think this should be nano's:
```suggestion
                if isinstance(v, py_datetime.datetime):
                    v = datetime.datetime_to_nanos(v)
```

Reply:
Haha - I meant to leave a comment here. Will do so we aren't confused reading this in the future